solutions/yarns_solution.md

```diff
@@ -0,0 +1,15 @@
+Solution to the two burning yarns
+======================================
+
+If you burn one piece from two ends, it takes 30 seconds to burn. 
+
+Now, you can burn one of them from two sides, and at the same time, burn one side of the other one. 
+When the first one is over (after 30 seconds), you burn the other side of the second one. This way you just counted 45 seconds.
+
+For the 50 seconds, you first burn one piece from two sides and wait until it is over. Now you just counted 30 seconds.
+At this moment, cut the second piece, into two pieces. 
+Burn the first one from both ends and the second one from one end. 
+You wait until one is over, and keep the process going while making sure at any moment, there are three flames that are burning. 
```

Comment:
The whole point is the yarns don't burn uniformly. 
(Please feel free to update the solution and problem statement in any way you think makes them more aligned with this GitHub repo.)

Count for 50 seconds:
At first, burn the first yarn from both sides. It takes 30 seconds for this yarn to burn. It means at, at the very moment the first yarn is fully burnt, you have counted 30 seconds. 

Now, you can use the second yarn to count for 20 seconds. 

If you burn the second yarn in such a way that at any moment in time, you have three flames going, burning this yarn takes 20 seconds. Therefore, you have counted 50 seconds in total. 

Let me know if there is anything I can help with.

Reply:
Oh okay, now I understand! Basically, what you are saying is by burning the second yarn from both ends _and_ also start burning it somewhere in the middle, it will take 60 / 3 = 20 seconds in total (if you somehow can maintain three flames at all times. Burning from the middle could result in two flames, each going in one direction. I suppose you then have to put out one of the flames. Again, too many uncertain variables in this riddle... :) ).

However, that surely requires the yarns to burn uniformly, right? There is no way you can count 30 seconds if both flames on the ends doesn't meet in the middle after 30 seconds (i.e. burns uniformly, with a constant rate)? For example, take the extreme case that the yarn burns in such a away that the flame is (almost) stationary at one end for 59 seconds and on the last second propagates through the whole yarn and burns it up. Burning this from two ends will not help you count 30 seconds.

Reply:
Great questions,
For the first part of the question, when you start dealing with the second yarn:
You first cut the second yarn somewhere in the middle. Now you have two pieces, therefore 4 ends. You burn 3 of those 4 ends. At some point, one of these pieces is going to be burnt out. At this time, you are left with one piece. 
What you would do with this piece? the similar thing you have done with the second yarn when you started dealing with it. You cut it somewhere in the middle and keep three flames going all the time.

Regarding the second question. There is no need for the yarns to burn uniformly. The whole yarn, if burnt from one side (left side), it takes 1 minute for the yarn to burn. If burnt from the opposite side (right side), it again takes 1 minute. If you burn it from both sides, after 15 seconds, the left side burns 15 seconds of its portion, and the right one does 15 seconds of its portion. After 30 seconds, they have to meet, otherwise, it results in a contradiction. 

I also thought this riddle may cause some ambiguities, I am not sure it is worth adding it if it causes some ambiguity. Let's remove it then. I will add some more concrete riddles in the feature. 
Cheers,